packages/core/mesh/network-manager/src/swarm/peer.ts

```diff
@@ -287,19 +292,19 @@ export class Peer {
       });
 
       // Calls `onStateChange` with CLOSED state.
-      void this.closeConnection(err);
+      void this.closeConnection(connection, err);
     });
 
     this.connection = connection;
 
     return connection;
   }
 
-  async closeConnection(err?: Error) {
-    if (!this.connection) {
+  async closeConnection(connection?: Connection, err?: Error) {
+    if (!connection) {
       return;
     }
-    const connection = this.connection;
+
```

Comment:
at end of function set this._connection to undefined.

Reply:
in that case, we won't be able to know if this was the main connection in onClose handler, around https://github.com/dxos/dxos/blob/4fe7b20df48e87ba51808783abf2ce18b617dec4/packages/core/mesh/network-manager/src/swarm/peer.ts#L255